components/kyma-environment-broker/internal/orchestration/handlers/cancel.go

```diff
@@ -0,0 +1,59 @@
+package handlers
+
+import (
+	orchestrationExt "github.com/kyma-project/control-plane/components/kyma-environment-broker/common/orchestration"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/dberr"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/dbsession/dbmodel"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Canceler struct {
+	orchestrations storage.Orchestrations
+	operations     storage.Operations
+	log            logrus.FieldLogger
+}
+
+func NewCanceler(operations storage.Operations, orchestrations storage.Orchestrations, logger logrus.FieldLogger) *Canceler {
+	return &Canceler{
+		orchestrations: orchestrations,
+		operations:     operations,
+		log:            logger,
+	}
+}
+
+// Cancel finds in progress orchestration and cancels it
+func (c *Canceler) Cancel() (string, error) {
+	orchestrations, _, _, err := c.orchestrations.List(dbmodel.OrchestrationFilter{States: []string{orchestrationExt.InProgress}})
```

Comment:
```suggestion
	orchestrations, _, size, err := c.orchestrations.List(dbmodel.OrchestrationFilter{States: []string{orchestrationExt.InProgress}})
```
Can we use returned parameter here? We already return size of the returned Orcherstration list